### PR TITLE
doc: Update links in docs

### LINKED ISSUE #4226: update osm_components_interactions.md
osm/docs/development_guide/osm_components_interactions.md needs to be updated as it is no longer the correct representation of how the components in osm interact

### PR DESCRIPTION
**Pull request by allenlsy**
**Description**:

Update links to source code files in the OSM Components & Interactions and Design
documents.

Diagram link:
* View only: https://drive.google.com/file/d/102PcUyFgm36l75e0ILxyeKq9Uu35OIGP/view?usp=sharing
* Edit: https://app.diagrams.net/#G102PcUyFgm36l75e0ILxyeKq9Uu35OIGP

Resolves #4226 

**Testing done**:

There is no code change.

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [X] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [ ] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

No

2. Is this a breaking change?

No

3. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)?

No

### CODE REVIEW

File: docs/development_guide/osm_components_interactions.md [hunk 1]
```diff
@@ -11,21 +11,22 @@ weight: 1
 ### Containers
```
Comment (snehachhabria) [inline docs/development_guide/osm_components_interactions.md:11]: Since this is a duplicate lets delete this, please

**Review by snehachhabria — COMMENTED** [review]
This doesn't resolve the issue; the diagram isn't a true representation of OSM anymore and needs to be updated. That the requirement of the issue

File: docs/development_guide/osm_components_interactions.md (hunk 1 above)
Comment (allenlsy) [reply docs/development_guide/osm_components_interactions.md:11]: Initially I thought the this file is still used by official doc site. Seems that's not the case. I will delete it.

**Comment by allenlsy** [comment]
@snehachhabria Diagram updated per feedback.  PTAL.

File: docs/development_guide/osm_components_interactions.md [hunk 2]
```diff
@@ -11,21 +11,22 @@ weight: 1
...
 
 When a new Pod creation is initiated, OSM's
-[MutatingWebhookConfiguration](https://github.com/openservicemesh/osm/blob/release-v0.3/charts/osm/templates/mutatingwebhook.yaml)
+(MutatingWebhookConfiguration)[https://github.com/openservicemesh/osm/blob/release-v1.0/pkg/injector/webhook.go#L422]
```
Comment (snehachhabria) [inline docs/development_guide/osm_components_interactions.md:14]: let's remove this page please, as it's a duplicate

File: DESIGN.md [hunk 3]
```diff
@@ -36,28 +36,28 @@ OSM ships out-of-the-box with all necessary components to deploy a complete serv
 
 
 ## OSM Components & Interactions
-![OSM Components & Interactions](/docs/development_guide/osm-components-and-interactions.png)
+![OSM Components & Interactions](/docs/images/osm-components-and-interactions.png)
```
Comment (snehachhabria) [inline DESIGN.md:39]: Once the page is removed, this reference can be removed as well

File: DESIGN.md [hunk 4]
```diff
@@ -36,28 +36,28 @@ OSM ships out-of-the-box with all necessary components to deploy a complete serv
...
 
 ### Containers
 When a new Pod creation is initiated, OSM's
-[MutatingWebhookConfiguration](/charts/osm/templates/mutatingwebhook.yaml)
+[MutatingWebhookConfiguration](/pkg/injector/webhook.go#L422)
```
Comment (snehachhabria) [inline DESIGN.md:43]: I don't think we should be referencing specific lines in the docs; these can change very frequently, and we may or may not remember to update them. Please remove exact line references in the code. If you wish to keep references, lets point to specific files and not lines

Comment (snehachhabria) [reply DESIGN.md:43]: Applies to everything below as well

**Review by snehachhabria — COMMENTED** [review]
A few suggestions on the diagram:
- Tresor is missing as one of the cert managers 
- we can indicate that the mesh config is created from osm-bootstrap
- osm-injector creates the sidecar on the k8s pod, this needs to be indicated

File: DESIGN.md (hunk 3 above)
Comment (allenlsy) [reply DESIGN.md:39]: This is the link to the image, rather than the removed page. We will keep this link.

**Review by snehachhabria — APPROVED** [review]

File: DESIGN.md [hunk 5]
```diff
@@ -36,28 +36,10 @@ OSM ships out-of-the-box with all necessary components to deploy a complete serv
...
+When a new Pod creation is initiated, OSM's [MutatingWebhookConfiguration](/pkg/injector/webhook.go) intercepts the create pod operations for namespaces joined to the mesh, and forwards these API calls to the OSM control plane. OSM control plane augments ([patches](/pkg/injector/patch.go)) the Pod spec with 2 new containers. One is the Envoy sidecar, the other is an init container. The init container is ephemeral. It executes a [generated `iptables` script](/pkg/injector/iptables.go) and terminates. The init container requires [NET_ADMIN Kernel capability](/pkg/injector/init_container.go) for [iptables](https://en.wikipedia.org/wiki/Iptables) changes to be applied. OSM uses `iptables` to ensure that all inbound and outbound traffic flows through the Envoy sidecar. The [init container Docker image](https://hub.docker.com/r/openservicemesh/init) is passed as a string pointing to a container registry. This is passed via the `spec.sidecar.initContainerImage` field of the `MeshConfig`. The default value is defined in the [chart values](/charts/osm/values.yaml).
```
Comment (shashankram) [inline DESIGN.md:42]:
```suggestion
When a new Pod creation is initiated, OSM's [Sidecar Injector](/pkg/injector/webhook.go) intercepts the create pod operations for namespaces joined to the mesh, and forwards these API calls to the OSM control plane. OSM control plane [patches](/pkg/injector/patch.go) the Pod spec with two new containers - 1. Init container, 2. Envoy sidecar. The init container is ephemeral. It executes a [set of `iptables` commands](/pkg/injector/iptables.go) and terminates. The init container requires [NET_ADMIN Kernel capability](/pkg/injector/init_container.go) for [iptables](https://en.wikipedia.org/wiki/Iptables) changes to be applied. OSM uses `iptables` to ensure that all inbound and outbound traffic is intercepted and redirected to the Envoy sidecar. The [init container Docker image](https://hub.docker.com/r/openservicemesh/init) is passed as a string pointing to a container registry. This is passed via the `spec.sidecar.initContainerImage` field of the `MeshConfig`. The default value is defined in the [chart values](/charts/osm/values.yaml).
```

**Comment by shashankram** [comment]
The image updates look good @allenlsy. If you used Google doc to create the image, would you mind sharing a link to the google doc hosting the image just in case we need to update it in the future? Was discussing with @trstringer about how to keep the images up to date, and we felt it would be best to use Google doc/slides to create images so we can easily edit them in the future.

**Review by shashankram — APPROVED** [review]

**Review by jaellio — APPROVED** [review]

**Pull request merged**